### PR TITLE
Add admonitions for unstable functionality

### PR DESCRIPTION
**Pull request by namurphy**
This PR adds and applies a reStructuredText substitution containing an admonition for functionality that is currently in development, which warns that API changes may occur in the future.

### CODE REVIEW

File: docs/plasma/index.rst
Comment (namurphy) [inline docs/plasma/index.rst:1]: - [ ] Revert changes to this file

File: docs/plasma/index.rst [hunk 1]
```diff
@@ -65,11 +69,6 @@ provide a method, which describes to the
 `Plasma <plasmapy.plasma.plasma_factory.PlasmaFactory>` factory
 which the data match its plasma data structure.
```
Comment (namurphy) [inline docs/plasma/index.rst:71]:
```suggestion
.. automodapi:: plasmapy.plasma
   :noindex:
   :no-main-docstring:
   :exclude-groups: modules, variables
```

File: docs/plasma/index.rst [hunk 2]
```diff
@@ -65,6 +69,11 @@ provide a method, which describes to the
...
 
+.. automodapi:: plasmapy.plasma
+   :noindex:
+   :no-main-docstring:
+   :exclude-groups: modules, variables
+
```
Comment (namurphy) [inline docs/plasma/index.rst:76]:
```suggestion
```

File: docs/plasma/index.rst [hunk 3]
```diff
@@ -165,6 +169,3 @@ API
 ---
 
 .. automodapi:: plasmapy.plasma
```
Comment (namurphy) [inline docs/plasma/index.rst:171]:
```suggestion
.. automodapi:: plasmapy.plasma
   :noindex:
   :no-main-docstring:
   :heading-chars: "^~"
```

**Review by ejohnson-96 — APPROVED** [review]
👍

**Pull request merged**